src/pages/iou/request/MoneyTemporaryForRefactorRequestParticipantsSelector.js

```diff
@@ -0,0 +1,384 @@
+
+    /**
+     * Adds a single participant to the request
+     *
+     * @param {Object} option
+     */
+    const addSingleParticipant = (option) => {
+        onParticipantsAdded([
+            {
+                ..._.pick(option, 'accountID', 'login', 'isPolicyExpenseChat', 'reportID', 'searchText'),
+                selected: true,
+            },
+        ]);
+        onFinish();
+    };
+
+    /**
+     * Removes a selected option from list if already selected. If not already selected add this option to the list.
+     * @param {Object} option
+     */
+    const addParticipantToSelection = useCallback(
+        (option) => {
+            const isOptionSelected = (selectedOption) => {
+                if (selectedOption.accountID && selectedOption.accountID === option.accountID) {
+                    return true;
+                }
+
+                if (selectedOption.reportID && selectedOption.reportID === option.reportID) {
+                    return true;
+                }
+
+                return false;
+            };
+            const isOptionInList = _.some(participants, isOptionSelected);
+            let newSelectedOptions;
+
+            if (isOptionInList) {
+                newSelectedOptions = _.reject(participants, isOptionSelected);
+            } else {
+                newSelectedOptions = [
+                    ...participants,
+                    {
+                        accountID: option.accountID,
+                        login: option.login,
+                        isPolicyExpenseChat: option.isPolicyExpenseChat,
+                        reportID: option.reportID,
+                        selected: true,
+                        searchText: option.searchText,
+                    },
+                ];
+            }
+
+            onParticipantsAdded(newSelectedOptions);
+
+            const chatOptions = OptionsListUtils.getFilteredOptions(
+                reports,
+                personalDetails,
+                betas,
+                isOptionInList ? searchTerm : '',
+                newSelectedOptions,
+                CONST.EXPENSIFY_EMAILS,
+
+                // If we are using this component in the "Request money" flow then we pass the includeOwnedWorkspaceChats argument so that the current user
+                // sees the option to request money from their admin on their own Workspace Chat.
+                iouType === CONST.IOU.TYPE.REQUEST,
+
+                // We don't want to include any P2P options like personal details or reports that are not workspace chats for certain features.
+                iouType !== CONST.IOU.REQUEST_TYPE.DISTANCE,
+            );
+
+            setNewChatOptions({
+                recentReports: chatOptions.recentReports,
+                personalDetails: chatOptions.personalDetails,
+                userToInvite: chatOptions.userToInvite,
+            });
```

Comment:
Coming from https://github.com/Expensify/App/issues/32819:
This code was redundant. More details in [PR](https://github.com/Expensify/App/pull/32847) description